sway/tree/container.c

```diff
@@ -1286,3 +1271,67 @@ bool container_is_fullscreen_or_child(struct sway_container *container) {
 
 	return false;
 }
+
+static void surface_send_enter_iterator(struct wlr_surface *surface,
+		int x, int y, void *data) {
+	struct wlr_output *wlr_output = data;
+	wlr_surface_send_enter(surface, wlr_output);
+}
+
+static void surface_send_leave_iterator(struct wlr_surface *surface,
+		int x, int y, void *data) {
+	struct wlr_output *wlr_output = data;
+	wlr_surface_send_leave(surface, wlr_output);
+}
+
+void container_discover_outputs(struct sway_container *con) {
+	if (!sway_assert(con->type == C_CONTAINER || con->type == C_VIEW,
+				"Expected a container or view")) {
+		return;
+	}
+	struct wlr_box con_box = {
+		.x = con->current.swayc_x,
+		.y = con->current.swayc_y,
+		.width = con->current.swayc_width,
+		.height = con->current.swayc_height,
+	};
+	struct sway_output *old_output = container_get_effective_output(con);
+
+	for (int i = 0; i < root_container.children->length; ++i) {
+		struct sway_container *output = root_container.children->items[i];
+		struct sway_output *sway_output = output->sway_output;
+		struct wlr_box output_box;
+		container_get_box(output, &output_box);
+		struct wlr_box intersection;
+		bool intersects =
+			wlr_box_intersection(&con_box, &output_box, &intersection);
+		int index = list_find(con->outputs, sway_output);
+
+		if (intersects && index == -1) {
+			// Send enter
+			wlr_log(WLR_DEBUG, "Con %p entered output %p", con, sway_output);
+			if (con->type == C_VIEW) {
```

Comment:
Do we really need to maintain a list of entered outputs per non-view container?

Reply:
Yes. If the container renders its own title (eg. a nested split) then the title texture needs to be recreated at the new scale when the container enters a new output.

Remember, the plan is for root, outputs and workspaces to not be part of `sway_container` any more, so the `outputs` property won't apply to them.